### PR TITLE
Add semver support

### PR DESCRIPTION
**Pull request by evgdem16**
Changelog: feature

### CODE REVIEW

File: docs/SEMVER_example [hunk 1]
```diff
@@ -0,0 +1,3 @@
+1.0.0
+29/08/2023 08:05:21
```
Comment (Abel-ze-Normand) [inline docs/SEMVER_example]: Why use custom formatted datetime string and implement custom parser for it? I think there should be enough of RFC3339 formated dt string.

File: src/healthcheck_httpd.erl [hunk 2]
```diff
@@ -77,22 +77,43 @@ init(Req, Opts) ->
 %%% private
 %%%===================================================================
 
-read_calver(Path) ->
+read_semver_file(Path, Number) ->
```
Comment (Abel-ze-Normand) [inline src/healthcheck_httpd.erl]: Pls refactor file reading operations. Currently this implementation re-reads version file 3 times for each /healthcheck call. This should be a single call for SEMVER file per requrest.

File: src/healthcheck_httpd.erl [hunk 3]
```diff
@@ -77,22 +77,43 @@ init(Req, Opts) ->
...
+                <<"(\\d{1,}).(\\d{1,}).(\\d{1,})-(\\S{0,}|$)">>,
                 [{capture, all_but_first, list}]) of
-        {match, [Year, Mounth, Day, Description]} ->
-                Year ++ "." ++ Mounth ++ "." ++ Day ++ "." ++ Description;
+        {match, [Major, Minor, Patch, Label]} ->
+                Major ++ "." ++ Minor ++ "." ++ Patch ++ "-" ++ Label;
         nomatch -> "undefined"
     end.
 
+find_version_datetime(Path) ->
```
Comment (Abel-ze-Normand) [inline src/healthcheck_httpd.erl]: Consider my proposal with using RFC3339 format

File: src/healthcheck_httpd.erl [hunk 4]
```diff
@@ -77,22 +77,43 @@ init(Req, Opts) ->
...
+    case re:run(binary_to_list(Data),
+                <<"(\\d{1,2})/(\\d{1,2}).(\\d{2,4}) (\\d{2}):(\\d{2}):(\\d{2})">>,
+                [{capture, all_but_first, list}]) of
+        {match, [Day, Month, Year, Hours, Minutes, Seconds]} ->
+                Day ++ "/" ++ Month ++ "/" ++ Year ++ " " ++ Hours ++ ":" Minutes ++ ":" Seconds;
+        nomatch -> "undefined"
+    end.
+
+
+find_service_name(Path) ->
```
Comment (Abel-ze-Normand) [inline src/healthcheck_httpd.erl]: Do we need this validation here? SEMVER file has been generated automatically, I bet it's not gonna be modified by hand, probably just get 3rd line from SEMVER and that's it?

**Review by Abel-ze-Normand — CHANGES_REQUESTED** [review]

File: src/healthcheck_httpd.erl (hunk 4 above)
Comment (evgdem16) [reply src/healthcheck_httpd.erl]: fixed

File: src/healthcheck_httpd.erl (hunk 3 above)
Comment (evgdem16) [reply src/healthcheck_httpd.erl]: fixed

File: src/healthcheck_httpd.erl (hunk 2 above)
Comment (evgdem16) [reply src/healthcheck_httpd.erl]: fixed

File: docs/SEMVER_example (hunk 1 above)
Comment (evgdem16) [reply docs/SEMVER_example]: fixed

**Review by Abel-ze-Normand — APPROVED** [review]
LGTM

**Pull request merged**